drivers/lcd/jd9851.c

```diff
@@ -0,0 +1,914 @@
+  jd9851_cmddata(dev, JD9851_SETRGBCYC, (const uint8_t *)
+                 "\x08\x00\x0a\x10\x08\x54\x45\x71\x2c", 9);
+  jd9851_cmddata(dev, JD9851_SETTCON, (const uint8_t *)
+                 "\x00\xa0\x79\x0e\x0a\x16\x79\x0e\x0a\x16\x79\x0e\x0a\x16\
+                  \x82\x00\x03", 17);
+  jd9851_cmddata(dev, JD9851_SETGD, (const uint8_t *)
+                 "\x04\x0c\x6b\x0f\x07\x03", 6);
+  jd9851_cmddata(dev, JD9851_SETRGBIF, (const uint8_t *) "\x22\x20", 2);
+  jd9851_cmddata(dev, JD9851_RAM_CTRL, (const uint8_t *) "\x00\x00", 2);
+
+  jd9851_cmddata(dev, JD9851_SET_PAGE_CMD, (const uint8_t *) "\x02", 1);
+  jd9851_cmddata(dev, JD9851_DCDC_SET2, (const uint8_t *)
+                 "\x19\xa0\x2f\x04\x33", 5);
+  jd9851_cmddata(dev, JD9851_SETPANEL, (const uint8_t *)
+                 "\x10\x66\x66\x01", 4);
+  jd9851_cmddata(dev, JD9851_OSCM_SET, (const uint8_t *) "\x01\x00\x00", 3);
+  jd9851_cmddata(dev, JD9851_SETMIPI_2, (const uint8_t *) "\x10\x20\xf4", 3);
+
+  jd9851_sendcmd(dev, JD9851_SLEEP_OUT);
+  up_mdelay(120);
+}
+
+/****************************************************************************
+ * Name: jd9851_sleep
+ *
+ * Description:
+ *   Sleep or wake up the driver.
+ *
+ ****************************************************************************/
+
+static void jd9851_sleep(FAR struct jd9851_dev_s *dev, bool sleep)
+{
+  if (sleep)
+    {
+      jd9851_sendcmd(dev, JD9851_SLEEP_IN);
+    }
+  else
+    {
+      jd9851_sendcmd(dev, JD9851_SLEEP_OUT);
+    }
+
+  up_mdelay(120);
+}
+
+/****************************************************************************
+ * Name: jd9851_display
+ *
+ * Description:
+ *   Turn on or off the display.
+ *
+ ****************************************************************************/
+
+static void jd9851_display(FAR struct jd9851_dev_s *dev, bool on)
+{
+  uint8_t reg;
+
+#if defined(CONFIG_LCD_JD9851_TE)
+  reg = JD9851_TE_MODE0;
+  jd9851_cmddata(dev, JD9851_TEON, &reg, 1);
+#endif
+
+#if defined(CONFIG_LCD_JD9851_INVERT)
+  jd9851_sendcmd(dev, JD9851_INVERSION_ON);
+#endif
+
+  if (on)
+    {
+      jd9851_sendcmd(dev, JD9851_DISON);
+    }
+  else
+    {
+      jd9851_sendcmd(dev, JD9851_DISOFF);
+    }
+}
+
+/****************************************************************************
+ * Name: jd9851_setorientation
+ *
+ * Description:
+ *   Set screen orientation.
+ *
+ ****************************************************************************/
+
+static void jd9851_setorientation(FAR struct jd9851_dev_s *dev)
+{
+  uint8_t reg = 0x00;
+
+#  if defined(CONFIG_LCD_RLANDSCAPE)
+
+  reg = JD9851_MADCTL_MY | JD9851_MADCTL_MV;
+
+#  elif defined(CONFIG_LCD_LANDSCAPE)
+
+  reg = JD9851_MADCTL_MV | JD9851_MADCTL_MX;
+
+#  elif defined(CONFIG_LCD_PORTRAIT)
+
+#  elif defined(CONFIG_LCD_RPORTRAIT)
+
+  reg = JD9851_MADCTL_MY | JD9851_MADCTL_MX;
+
+#  endif
+
+#  if defined(CONFIG_LCD_JD9851_BGR)
+
+  reg |= JD9851_MADCTL_BGR;
+
+#  endif
```

Comment:
Please fix spacing after "#" on these #if / #elif / #endif

Reply:
not verified on JD9852, I can update later when got it.